core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -199,4 +220,56 @@ public void validate(TableMetadata base) {
       }
     }
   }
+
+  class AssertLastAssignedViewVersionId implements UpdateRequirement {
+    private final int versionId;
+
+    public AssertLastAssignedViewVersionId(int versionId) {
+      this.versionId = versionId;
+    }
+
+    public int versionId() {
+      return versionId;
+    }
+
+    @Override
+    public void validate(TableMetadata base) {
+      // noop
+    }
+
+    @Override
+    public void validate(ViewMetadata base) {
+      if (versionId != base.currentVersionId()) {
+        throw new CommitFailedException(
+            "Requirement failed: last assigned view version id changed: expected id %s != %s",
+            versionId, base.currentVersionId());
+      }
+    }
+  }
+
+  class AssertCurrentViewVersionID implements UpdateRequirement {
```

Comment:
Do we need this?

I don't think we need it because adding a new view version will assign a new ID (see the comment above) and setting the current version is an idempotent operation. We have a similar assertion, `AssertRefSnapshotId`, for snapshots, but that is because snapshots are based on one another. Here, view versions are completely independent and so replacing is an idempotent operation.

We _could_ add this if we wanted to make it possible to perform either an idempotent replace OR fail. But I don't really see a good argument for the failure case. Open to suggestion here though.

Reply:
> We could add this if we wanted to make it possible to perform either an idempotent replace OR fail.

Yes that was exactly the goal here and the only reason we would need this. However, I can't find a strong argument that would justify having this now. 

That being said, I think you're right and we can remove all requirements for Views for now.